user_tools/src/spark_rapids_pytools/common/utilities.py

```diff
@@ -229,6 +230,40 @@ def get_and_setup_logger(cls, type_label: str, debug_mode: bool = False):
             logger.addHandler(fh)
         return logger
 
+    @classmethod
+    def modify_log4j_properties(cls, prop_file_path: str, new_log_file_path: str) -> str:
+        """
+        Modifies the log file path in a log4j properties file to redirect logging output to a new location.
+
+        This method reads an existing log4j.properties file and alters the log file path specified
+        for the FILE appender. The modified properties file is saved to a temporary file,
+        which is returned to the caller to be used as the new log4j configuration. This temporary file
+        is deleted after the java process is completed.
+
+        :param prop_file_path: The file path to the original log4j.properties file. This file
+                               should contain configurations for the log4j logging utility.
+        :param new_log_file_path: The base output directory where the new log file will be created.
```

Comment:
nit: is it better to name this argument to something like `new_log_dir`, otherwise `new_log_file_path` sounds more like a path than a directory?